### PR TITLE
Support PPE profiling for `S3`

### PR DESCRIPTION
**Pull request by KantaTamura**
This PR resolves the `S3` task in #258 

Sample program to check tracing
```
import json
import pytorch_pfn_extras as ppe

from pfio.v2 import S3, Path, from_url

def p():
    tracer = ppe.profiler.get_tracer()
    tracer.clear()

    fil = 0

    with from_url('pfs1://cluster-tutorial/', trace=True) as fs:
        assert isinstance(fs, S3)

        for f in fs.list(recursive=True):
            if f == "imagenet-classification.zip":
                continue

            with fs.open(f, 'rb') as fp:
                tmp = fp.read()

            fil += 1

            if fil > 10:
                break

    print(fil, "files")

    dict = tracer.state_dict()
    keys = [event["name"] for event in json.loads(dict["_event_list"])]
    print(keys)

    w = ppe.writing.SimpleWriter(out_dir="")
    tracer.initialize_writer("trace_s3.json", w)
    tracer.flush("trace_s3.json", w)

p()
```

output
```
11 files
['pfio.v2.S3:list-0', 'pfio.v2.S3:open', 'pfio.v2.S3:read', 'pfio.v2.S3:exit-context', 'pfio.v2.S3:list-1', 'pfio.v2.S3:open', 'pfio.v2.S3:read', 'pfio.v2.S3:exit-context', 'pfio.v2.S3:list-2', 'pfio.v2.S3:open', 'pfio.v2.S3:read', 'pfio.v2.S3:exit-context', 'pfio.v2.S3:list-3', 'pfio.v2.S3:open', 'pfio.v2.S3:read', 'pfio.v2.S3:exit-context', 'pfio.v2.S3:list-4', 'pfio.v2.S3:open', 'pfio.v2.S3:read', 'pfio.v2.S3:exit-context', 'pfio.v2.S3:list-5', 'pfio.v2.S3:open', 'pfio.v2.S3:read', 'pfio.v2.S3:exit-context', 'pfio.v2.S3:list-6', 'pfio.v2.S3:open', 'pfio.v2.S3:read', 'pfio.v2.S3:exit-context', 'pfio.v2.S3:list-7', 'pfio.v2.S3:open', 'pfio.v2.S3:read', 'pfio.v2.S3:exit-context', 'pfio.v2.S3:list-8', 'pfio.v2.S3:open', 'pfio.v2.S3:read', 'pfio.v2.S3:exit-context', 'pfio.v2.S3:list-9', 'pfio.v2.S3:open', 'pfio.v2.S3:read', 'pfio.v2.S3:exit-context', 'pfio.v2.S3:list-10', 'pfio.v2.S3:open', 'pfio.v2.S3:read', 'pfio.v2.S3:exit-context', 'pfio.v2.S3:list-11']
```

rendering of the output json file (trace.json) with chrome://tracing is shown below
image

### CODE REVIEW

**Comment by KantaTamura** [comment]
add profiling `boto3` api
image

File: pfio/v2/s3.py [hunk 1]
```diff
@@ -380,7 +423,8 @@ def _reset(self):
 
     def _connect(self):
         # print('boto3.client options:', kwargs)
-        self.client = boto3.client('s3', **self.kwargs)
+        obj = boto3.client('s3', **self.kwargs)
+        self.client = Boto3ProfileWrapper(obj)
```
Comment (k5342) [inline pfio/v2/s3.py:427]: It's better to switch here between the vanilla boto3 and the wrapped boto3 client by `self.trace` as well as `Local` fs does.
https://github.com/pfnet/pfio/blob/9bf5900d1aa8403633eb9a15af957aa11ff80a19/pfio/v2/local.py#L123-L127

**Review by k5342 — CHANGES_REQUESTED** [review]
Overall, it looks good. I have one minor suggestion.

File: pfio/v2/s3.py (hunk 1 above)
Comment (KantaTamura) [reply pfio/v2/s3.py:427]: Forgot to change from test code ...

Comment (KantaTamura) [reply pfio/v2/s3.py]: fix and rebase `master` branch

**Comment by k5342** [comment]
PR #343 merge raised a conflict. Could you resolve the conflict before merging this, please?

**Review by k5342 — APPROVED** [review]

**Pull request merged**